Add MIG `atesting_a100` documentation (#400)
ResearchComputing/Documentation: docs/clusters/alpine/alpine-hardware.md

#### Special-Purpose Partitions

To help users test out their workflows, CURC provides several special-purpose partitions on Alpine. These testing partitions allow users to gain access to resources more quickly than the other partitions. This access allows users the ability to test or compile code on CPU and GPU compute nodes. To ensure equal access to these resources, the amount of resources (such as CPUs, GPUs, and runtime) are limited. 

```{important}

> All compile and testing partitions are not meant to run non-testing workflows. Please utilize the appropriate partitions when running code. 

Reviewer: ```suggestion
Compiling and testing partitions are, as their name implies, only meant for compiling code and testing workflows. They are not to be used outside of compiling or testing. Please utilize the appropriate partitions when running code. 
```

````
`````

##### GPU `atesting` usage examples:

`atesting_a100` and `atesting_mi100` provide access to limited GPU resources for the purpose of verifying GPU workflows and building GPU-accelerated applications. For the `atesting_mi100` partition, users can request up to 3 GPUs and all associated CPU cores (64 max) from a single node for up to one hour. Due to limitations with MIG (see below), we limit users to 1 GPU (with 20 GB of VRAM) and at most 10 CPU cores on the `atesting_a100` partition.

```{important}

> The `atesting_a100` partition utilizes NVIDIA's [Multi-Instance GPU (MIG)](https://docs.nvidia.com/datacenter/tesla/mig-user-guide/index.html) feature. MIG is fantastic because it allows us to slice our GPUs into multiple instances, thus providing more usable GPUs on our system. Although MIG has tremendous advantages on our `atesting_a100` partition, there are certain limitations to MIG. One important limitation is that MIG does not allow for multiple MIG slices to communicate with each other. This is the reason we limit users to just 1 GPU on the `atesting_a100` partition. For more information on limitations of MIG, please see NVIDIA's MIG [Application Considerations](https://docs.nvidia.com/datacenter/tesla/mig-user-guide/index.html#application-considerations) documentation. 

Reviewer: ```suggestion
The `atesting_a100` partition utilizes NVIDIA's [Multi-Instance GPU (MIG)](https://docs.nvidia.com/datacenter/tesla/mig-user-guide/index.html) feature which can "slice" GPUs into multiple instances. The MIG "slices" allows more users to share access of the GPUs on our system. But the increase in GPU access provided by MIG does come with certain limitations. One important limitation is that MIG does not allow for multiple MIG slices to communicate with each other. This is the reason we limit users to just 1 GPU on the `atesting_a100` partition. For more information on limitations of MIG, please see NVIDIA's MIG [Application Considerations](https://docs.nvidia.com/datacenter/tesla/mig-user-guide/index.html#application-considerations) documentation. 
```
